applications/conversations/controllers/class.messagescontroller.php

```diff
@@ -368,6 +368,14 @@ public function index($conversationID = false, $offset = -1, $limit = '') {
 
         $this->Conversation = $this->ConversationModel->getID($conversationID);
         $this->Conversation->Participants = $this->ConversationModel->getRecipients($conversationID);
+
+        //Verify if participant email should be visible
+        foreach ($this->Conversation->Participants as $participant) {
+            if (!Gdn::session()->checkPermission('Garden.PersonalInfo.View')) {
```

Comment:
You are going a permission check multiple times for no reason. The code should be:
```php
if (!Gdn::session()->checkPermission('Garden.PersonalInfo.View')) {
    foreach ($this->Conversation->Participants as $participant) {
        $participant->Email = '';
    }
}
```

The permission check is also wrong. It should be `Garden.Email.View`.

Reply:
`Garden.Email.View` determine if one can receive emails... `Garden.PersonalInfo.View` was correct